lib/models/project.js

```diff
@@ -105,23 +104,44 @@ class Project {
     logger.info('bowerDirectory: %s', this.bowerDirectory);
   }
 
-  hasDependencies() {
-    return !!this.nodeModulesPath;
+  // This exists only for backward compat, because some addons use
+  // it. It was never marked public and we can totally nuke it, but we
+  // can be polite and give them a warning for a while.
+  get nodeModulesPath() {
+    if (!this._nodeModulesPath) {
+      let nodeModulesPath = require('node-modules-path');
+      let value = nodeModulesPath(this.root);
+      this._nodeModulesPath = { value };
+      this.ui.writeDeprecateLine('An addon is trying to access project.nodeModulesPath. This is not a reliable way to discover npm modules. Instead, consider doing: require("resolve").sync(something, { basedir: project.root })');
+    }
+    return this._nodeModulesPath.value;
   }
 
-  /**
-    Sets the path to the node_modules directory for this
-    project.
-
-    @private
-    @method setupNodeModulesPath
-   */
-  setupNodeModulesPath() {
-    this.nodeModulesPath = nodeModulesPath(this.root);
-
-    logger.info('nodeModulesPath: %s', this.nodeModulesPath);
+  // Checks whether the project's npm dependencies are
+  // present. Previously this just looked for a node_modules folder in
+  // a fixed place (which is not compatible with node's module
+  // resolution algorithm). Now we just sample to see if we can
+  // resolve the first dependency or devDependency we find.
+  hasDependencies() {
+    if (this.cli.disableDependencyChecker) {
+      // Blueprint tests set this flag.
+      return true;
+    }
+    for (let depName in this.dependencies()) {
+      try {
+        this.resolveSync(depName);
+        return true;
```

Comment:
Looks like it will return true on the first found package. Is it intentional or should we check all the deps?

Reply:
Ah sorry, missed you comment few lines above